### PR TITLE
Fix Path to Carbon Black Solution after move

### PR DESCRIPTION
**Pull request by joelst**
Change(s):
   - Updated URLs for Azure deployment buttons to work since being moved from Playbooks to Solutions
   - Minor documentation updates for grammar
   - Moved Data Connector out of Playbooks sub folder.
   - Updated Solution Metadata format.
   - Added analytic rule entity mappings to `mainTemplate.json`
   - Added configuration steps to `createUiDefinition.json` to provide user with needed info.
 - Additional work can be done in the future to merge the Data Connector with this solution as well as general improvements and polish to the solution. To match wording of other solutions.
  
   Reason for Change(s):
   - `\Playbooks\CarbonBlack` was moved to `\Solutions\CarbonBlack` in #4340 breaking the following:
     - The VMware Carbon Black Solution fails to deploy from the content hub because of an invalid path.
     - Deploy to Azure buttons on the README files also have invalid paths and fail.
   
   Version Updated:
   - NA

   Testing Completed:
   - Yes

### CODE REVIEW

**Comment by joelst** [comment]
I updated the customUiDefinition to include information about prerequisites and post-deployment steps. This should reduce the need for referencing the README files in the repo. In the next PR, we can do a full cleanup of the legacy README files.

Please make sure all of the Solution json files are in the right place and correct or point me to a reference to verify.

**Comment by v-sabiraj** [comment]
@joelst, there are conflicts in the branch ,please resolve them, thanks.

**Comment by v-spadarthi** [comment]
@joelst, Could you please resolve the conflicts in the branch, thanks.

File: Sample Data/PublicFeeds/MITREATT&CK/MicrosoftSentinel.csv [hunk 1]
```diff
@@ -277779,4 +277779,4 @@ N.A.,N.A.,Azure,Analytics,Microsoft Built-in Alerts,N.A.,Unusual file share acti
 N.A.,N.A.,Azure,Analytics,Microsoft Built-in Alerts,N.A.,Multiple delete VM activities,N.A.,MicrosoftCloudAppSecurity,SecurityAlert (MCAS) | McasShadowItReporting,N.A.,N.A.,N.A.,N.A.,N.A.,N.A.,https://docs.microsoft.com/cloud-app-security/investigate-anomaly-alerts,2022-05-20
 N.A.,N.A.,Azure,Analytics,Microsoft Built-in Alerts,N.A.,Ransomware activity,N.A.,MicrosoftCloudAppSecurity,SecurityAlert (MCAS) | McasShadowItReporting,N.A.,N.A.,N.A.,N.A.,N.A.,N.A.,https://docs.microsoft.com/cloud-app-security/investigate-anomaly-alerts,2022-05-20
 N.A.,N.A.,Azure,Analytics,Microsoft Built-in Alerts,N.A.,Unusual file deletion activity (by user),N.A.,MicrosoftCloudAppSecurity,SecurityAlert (MCAS) | McasShadowItReporting,N.A.,N.A.,N.A.,N.A.,N.A.,N.A.,https://docs.microsoft.com/cloud-app-security/investigate-anomaly-alerts,2022-05-20
-N.A.,N.A.,Azure,Analytics,Microsoft Built-in Alerts,N.A.,Investigation priority score increase (preview),N.A.,MicrosoftCloudAppSecurity,SecurityAlert (MCAS) | McasShadowItReporting,N.A.,N.A.,N.A.,N.A.,N.A.,N.A.,https://docs.microsoft.com/cloud-app-security/investigate-anomaly-alerts,2022-05-20
```
Comment (joelst) [inline Sample Data/PublicFeeds/MITREATT&CK/MicrosoftSentinel.csv:277782]: This file was change by a github bot and not from my original pr. The file matches the file in master.

**Comment by joelst** [comment]
> Hey, @joelst, there is a file movement of azuredeploy.json from playbooks folder to data connector folder, actually that maybe the custom connector but not the data connector, please check and update, thanks.

I saw other solutions that put the Logic App connector in the Data Connectors folder, so I followed suit. I have now moved it to the Playbooks folder and updated all references. I apologize for not seeing your message earlier.

**Comment by joelst** [comment]
> @joelst, there are conflicts in the branch ,please resolve them, thanks.

There was a file updated by the github-actions bot that seemed to an issue. It was not a file that was changed in the original PR. All conflicts have been resolved.

**Comment by v-spadarthi** [comment]
@aprakash13 :Please review Analytics. Thanks!

**Comment by joelst** [comment]
> @rushriva @anki-narravula : Please review playbooks. Thanks.

Most of the changes were done in the README and deploy file to mimic newer deployments. There were also minor changes with names from Azure Sentinel to Microsoft Sentinel. Please let me know if you have any questions.

**Comment by joelst** [comment]
> @aprakash13 :Please review Analytics. Thanks!

Most of the changes were done in the README and deploy file to mimic newer deployments. There were also minor changes with names from Azure Sentinel to Microsoft Sentinel. Please let me know if you have any questions.

**Comment by joelst** [comment]
@NikTripathi @v-rucdu 

You created and approved PR #5743 today. I've had this PR open for three months that fixed the URL issue, but also fixes issues in the documentation. 

I'm not sure who needs to look at this to provide feedback so that I can fix it, or if I should abandon the PR.

**Comment by anki-narravula** [comment]
Hi @joelst - 
Please accept our apologies for not communicating the things clearly to you. Team was in migration mode and migrating all the old solutions to new template specs and missed out this. really sorry for this. 
At this point of time, I am not sure how you feel if I ask you something. But we don't want to lose the opportunity to improve this solution and we want your fixes for this solution definitely. To take this forward we need to make the changes on top of latest master and then only I can merge, can you please do this favor

**Comment by v-marimanda** [comment]
@joelst  Please resolve the conflicts. thanks!!!

**Comment by joelst** [comment]
I will look through this next week when I get back. Thanks for looking it.

**Comment by NikTripathi** [comment]
@joelst Ok Thanks.

**Comment by v-spadarthi** [comment]
@joelst Please resolve the conflicts. thanks!!!

**Comment by v-mchatla** [comment]
Hi @joelst,
As part of solution migration, we have renamed the carbon-black to [VMware Carbon Black Cloud](https://github.com/Azure/Azure-Sentinel/tree/master/Solutions/VMware%20Carbon%20Black%20Cloud). Please go through the files inside this folder and make changes to your PR accordingly. Sorry for the inconvenience caused.
Thanks

**Comment by v-mchatla** [comment]
Hi @joelst 
Can you please accommodate the requested changes, so that we can review and merge it.
Thanks

**Comment by joelst** [comment]
Since this is now so far out of date. I will create a new pull request with my changes once I have them reworked into the new folder structure.

Thanks!

**Comment by v-mchatla** [comment]
Hi @joelst,
We really appreciate your contribution to this repository. Earlier out team is working on migration activity, so we are not closely working on the PRs. But now we are focusing more on the PRs and get it addressed as soon as possible. 
Thanks

**Pull request closed without merging**